flox/aggregations.py

```diff
@@ -567,6 +568,98 @@ def quantile_new_dims_func(q) -> tuple[Dim]:
 mode = Aggregation(name="mode", fill_value=dtypes.NA, chunk=None, combine=None)
 nanmode = Aggregation(name="nanmode", fill_value=dtypes.NA, chunk=None, combine=None)
 
+
+@dataclass
+class Scan:
+    # This dataclass is separate from aggregations since there's not much in common
+    # between reductions and scans
+    name: str
+    # binary operation (e.g. add)
+    binary_op: Callable
+    # in-memory grouped scan function (e.g. cumsum)
+    scan: str
+    # Grouped reduction that yields the last result of the scan (e.g. sum)
+    reduction: str
+    # Identity element
+    identity: Any
+    # dtype of result
+    dtype: Any = None
+
+
+@dataclass
+class AlignedArrays:
+    """Simple Xarray DataArray type data class with two aligned arrays."""
+
+    array: np.array
+    group_idx: np.array
```

Comment:
```suggestion
    array: np.ndarray
    group_idx: np.ndarray
```
Should fix errors like: `error: np.array? has no attribute "ndim"  [attr-defined`